frontend/cypress/support/commands.ts

```diff
@@ -44,47 +44,49 @@ Cypress.Commands.add('login', (provider: string, username: string, password: str
+      // Cypress.Cookies.debug(true) // now Cypress will log when it alters cookies
+      // cy.getCookies()
 
-        cy.log(
-          `provider: ${provider}, 
+      cy.log(
+        `provider: ${provider},
 					username: ${username},
 					auth_strategy: ${auth_strategy}`
-        );
-        cy.visit('/');
+      );
+      // Disabling refresh as this can prevent navigation to another page
+      // until the overview page has fully loaded which can be greatly delayed
+      // if the refresh interval is lower and the api requests are slow.
+      cy.visit('/console/overview?duration=60&refresh=0');
 
-        if (auth_strategy === 'openshift') {
-          cy.get('.pf-c-form').contains(auth_strategy, { matchCase: false }).click();
-        }
-
-        if (auth_strategy === 'openshift') {
-          cy.get('.pf-c-button').contains(provider, { matchCase: false }).click();
-        }
-
-        if (auth_strategy === 'openshift') {
-          cy.get('#inputUsername').clear().type(username);
-          cy.get('#inputPassword').clear().type(password);
-          cy.get('button[type="submit"]').click();
-          cy.wait('@authorized').its('response.statusCode').should('eq', 200);
-          cy.getCookie('kiali-token-aes', { timeout: 15000 })
-            .should('exist')
-            .then(() => {
-              haveCookie = true;
-            });
-          // Wait for the redirect to the overview page after a successful login.
-          // Otherwise the redirect can mess with page loading on subsequent tests.
-          cy.contains('loading', {matchCase: false}).should('not.exist')
-          cy.url().should('include', '/overview');
-        }
-      } else {
-        cy.log('got an auth cookie, skipping login');
+      if (auth_strategy === 'openshift') {
+        cy.get('.pf-c-form').contains(auth_strategy, { matchCase: false }).click();
+        cy.get('.pf-c-button').contains(provider, { matchCase: false }).click();
+        cy.get('#inputUsername').clear().type(username);
+        cy.get('#inputPassword').clear().type(password);
+        cy.get('button[type="submit"]').click();
+        cy.wait('@authorized').its('response.statusCode').should('eq', 200);
+        cy.getCookie('kiali-token-aes', { timeout: 15000 })
+          .should('exist')
+          .then(() => {
+            haveCookie = true;
+          });
+        // Wait for the redirect to the overview page after a successful login.
+        // Otherwise the redirect can mess with page loading on subsequent tests.
+        cy.contains('loading', { matchCase: false }).should('not.exist');
+        cy.url().should('include', '/overview');
+        // Wait for the overview page to load some elements to be sure that things
+        // have settled. Otherwise some part of the overview page will change the
+        // url back to the overview page.
+        cy.getBySel('alpha-EXPAND');
       }
+    } else {
+      cy.log('got an auth cookie, skipping login');
```

Comment:
I need fresh build to verify this on OCP, will take a look tomorrow after fresh nightly. Right now everything fails on this step. 

I do install `hack/istio/install-testing-demos.sh` without any parameters, is that right? @nrfox @jmazzitelli

Reply:
Yes. And unfortunately, it seems like for every test run you should delete and then re-install the demos. I've seen errors when I run the cypress tests multiple times without resetting the demos between each run. I haven't figured out why - it is probably due to the fact the cypress tests do not "undo" things they might change during their run. We need these tests to leave the demos in the same state they were in when the tests start, but it seems like at least some tests do not do this.

Use `-d true` to tell the script to uninstall the demos:

```
hack/istio/install-testing-demos.sh -c oc -d true
```

Reply:
> We need these tests to leave the demos in the same state they were in when the tests start

If this is the case I have some solutions for that

p.s.: install demos script works fine with defaults and fallback to oc 👍